taf/tests/test_updater/test_update_library/test_clone_library_invalid.py

```diff
@@ -0,0 +1,220 @@
+        },
+    )
+    setup_manager.execute_tasks()
+
+    # Run the updater which will clone and then update
+    with pytest.raises(UpdateFailedError, match=CANNOT_CLONE_TARGET_PATTERN):
+        update_full_library(
+            library_with_dependencies,
+            origin_dir,
+            client_dir,
+            operation=OperationType.CLONE,
+            expected_repo_type=UpdateType.EITHER,
+            excluded_target_globs=None,
+        )
+        update_full_library(
+            library_with_dependencies,
+            origin_dir,
+            client_dir,
+            operation=OperationType.UPDATE,
+            expected_repo_type=UpdateType.EITHER,
+            excluded_target_globs=None,
+        )
+
+
+@pytest.mark.parametrize(
+    "library_with_dependencies",
+    [
+        {
+            "targets_config": [{"name": "target1"}, {"name": "target2"}],
+            "dependencies_config": [
+                {
+                    "name": "namespace1/auth",
+                    "targets_config": [
+                        {"name": "namespace1/target1"},
+                        {"name": "namespace1/target2"},
+                    ],
+                },
+                {
+                    "name": "namespace2/auth",
+                    "targets_config": [
+                        {"name": "namespace2/target1"},
+                        {"name": "namespace2/target2"},
+                    ],
+                },
+                # Add additional dependencies as needed
+            ],
+        },
+    ],
+    indirect=True,
+)
+def test_update_all_except_invalid(
+    library_with_dependencies,
+    origin_dir,
+    client_dir,
+):
+    # Invalidate one of the target repositories of a referenced authentication repository
+    auth_repo = library_with_dependencies["namespace1/auth"]["auth_repo"]
+    setup_manager = SetupManager(auth_repo)
+    setup_manager.add_task(
+        invalidate_target_repo,
+        kwargs={
+            "library_with_dependencies": library_with_dependencies,
+            "namespace": "namespace1/auth",
+            "target_name": "namespace1/target1",
+        },
+    )
+    setup_manager.execute_tasks()
+
+    # Run the updater which will clone and then update
+    with pytest.raises(UpdateFailedError, match=CANNOT_CLONE_TARGET_PATTERN):
+        update_full_library(
+            library_with_dependencies,
+            origin_dir,
+            client_dir,
+            operation=OperationType.CLONE,
+            expected_repo_type=UpdateType.EITHER,
+            excluded_target_globs=None,
+        )
+        update_full_library(
+            library_with_dependencies,
+            origin_dir,
+            client_dir,
+            operation=OperationType.UPDATE,
+            expected_repo_type=UpdateType.EITHER,
+            excluded_target_globs=None,
+        )
+
+    # Verify that the valid repositories were updated successfully
+    for auth_repo_name, repo_info in library_with_dependencies.items():
```

Comment:
I'd move this to a separate function (`update_utils` could be a place for it). We could check if all valid repositories were updated successfully in every invalid update test. What we should do here is save the head commit shas before the update, run update where we expect an update of one or more dependencies to fail, compare the commits. We should update at least two dependencies - push valid commits to one of them, push invalid commits to another one. After updating the whole library using the update. Then check if head commits shas of repositories to which we pushed valid commits got updated after the update (update the origin repo, the updater should update the client's repositories). This can be a separate issue and not part of this PR. Please create an issue if you'd rather move on to another task.